File: java/src/main/java/com/jay2u8809/codesamples/corp/uzjp/adakr/controller/qrcode/QrCodeController.java
package com.jay2u8809.codesamples.corp.uzjp.adakr.controller.qrcode;

import com.jay2u8809.codesamples.common.CommonExtends;
import com.jay2u8809.codesamples.corp.uzjp.adakr.api.qrcode.QrCodeApi;
import com.jay2u8809.codesamples.corp.uzjp.adakr.controller.qrcode.dto.QrCodeViewEntry;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;

@RequiredArgsConstructor
@Controller
@RequestMapping(value = "/corp/uzjp/adakr/qrcode")
public class QrCodeController extends CommonExtends {

    private static final String BASE_PATH = CORP_UZJP_HOME_PATH + "/adakr/qrcode/";

    private final QrCodeApi qrCodeApi;

    @GetMapping(value = "/entry/")
    public String moveGenerateQrCodeView (Model model) {

        // Form Url Data
        String formUrl = BASE_PATH + "generate/";
        model.addAttribute("formUrl", formUrl);

        return BASE_PATH + "generate_qrcode";
    }

    /**
     *
     * @param qrCodeViewEntry
     * @return
     */
    @PostMapping(value = "/generate/")
    @ResponseBody
    public ResponseEntity<?> generateQrCode (QrCodeViewEntry qrCodeViewEntry) {

        logger.debug("Input Uri : {}", qrCodeViewEntry.getQrcodeUri());

        Map<String, Object> result = new HashMap<>();
        String qrCode = qrCodeApi.generateQrCodeByImageUri(qrCodeViewEntry.getQrcodeUri(), qrCodeViewEntry.getQrcodeSize());
//        String qrCode = "QRCODE TEST URI";
        result.put("qrCodeUrl", qrCode);

        // Return Type : JSON
        return ResponseEntity.status(HttpStatus.OK).body(result);
    }

}

File: javascript/nodejs/nest-sls-sample-project/src/api/open/qrcode/qrcode.controller.ts
import { Body, Controller, Param, Post, Query } from '@nestjs/common';
import { ApiTags } from '@nestjs/swagger';
import { QrcodeRequestDto } from './dto/qrcode-request.dto';
import { QrcodeService } from './qrcode.service';

const TAG = 'QrcodeController';

@ApiTags('open/qrcode')
@Controller('/v1/qrcode')
export class QrcodeController {
  constructor(private readonly qrcodeService: QrcodeService) {}

  @Post('/generate')
  public generateQrCode(@Body() params: QrcodeRequestDto): string {
    console.log(TAG, 'req-params', JSON.stringify(params));
    return this.qrcodeService.generateQrCodeByImageUri(params.url, params.size);
  }
}
